src/gov/nist/javax/sip/stack/SIPDialog.java

```diff
@@ -1914,6 +1914,7 @@ public boolean addTransaction(SIPTransaction transaction) {
                 && !firstTransactionId.equals(transaction.getBranchId())
                 && transaction.getMethod().equals(firstTransactionMethod)) {
             setReInviteFlag(true);
+            ackProcessed = false;
```

Comment:
In the bug report (https://github.com/RestComm/jain-slee.sip/issues/36) there is a SIPP TC which was used to test this correction, but I can additionally create junit TC.